core/idl/wallet/ethereum/ethereum_like_wallet.djinni

```diff
@@ -1,10 +1,114 @@
 @import "../../errors.djinni"
+@import "../amount.djinni"
+@import "../../ethereum/addresses.djinni"
+@import "erc20.djinni"
 
 GetEthreumLikeWalletCallback = interface +j +o +s +n {
     onSuccess(wallet: EthereumLikeWallet, isCreated: bool);
     onError(error: Error);
 }
-
+                                                                                                                             
 EthereumLikeWallet = interface +c {
 
-}
\ No newline at end of file
+}
+
+#Class representing a Ethereum transaction
+EthereumLikeTransaction = interface +c {
+    # Get the hash of the transaction.
+    getHash(): string;
+    # Get the nonce of the transaction : sequence number issued by originating EOA
+    getNonce(): i32;
+    # Get Gas price (in wei)
+    getGasPrice(): Amount;
+    # Get start gas (in wei) : maximum amount of gas the originator is willing to pay
+    getGasLimit(): Amount;
+    # Effective used gas
+    getGasUsed(): Amount;
+    # Get destination ETH address
+    getReceiver(): EthereumLikeAddress;
+    # Get ETH sender address
+    getSender(): EthereumLikeAddress;
+    # Get amount of ether to send
+    getValue(): Amount;
+    # Get binary data payload
+    getData(): optional<binary>;
+    # Serialize the transaction to its raw format.
+    serialize(): binary;
+    # Set signature of transaction, when a signature is set serialize method gives back serialized Tx
+    setSignature(vSignature: binary, rSignature: binary, sSignature: binary);
+    setDERSignature(signature: binary);
+    setVSignature(vSignature: binary);
+    # Get the time when the transaction was issued or the time of the block including
+    # this transaction
+    getDate(): date;
+    # Get block to which transaction belongs (was mined in)
+    getBlock(): optional<EthereumLikeBlock>;
+}
+
+#Class representing a Ethereum Operation
+EthereumLikeOperation = interface +c {
+    #Get operation's transaction
+    #@return EthereumLikeTransaction object
```

Comment:
We should rework on documentation of idls anyway and should conform to a standard that we agree on and stick with (we can talk about it since you are working on doc also)

Reply:
Sure. I’m going to keep track on everything we need to do with documentation and I’ll create a big issue in Jira for that, if that’s okay for you — or gather already existing issues.